### PR TITLE
Added forum members only feature

### PR DESCRIPTION
**Pull request by sometimesok**
Am admin should be able to specify what section of the forums can only be viewed my registered users.

### CODE REVIEW

File: ISS/views/forum.py [hunk 1]
```diff
@@ -75,8 +76,8 @@ def thread_index(request, forum_id):
         'can_start_thread': forum.create_thread_pack.check_request(request)
     }
 
-    if request.user.is_authenticated:
-        forum.mark_read(request.user)
+    #if request.user.is_authenticated:
+    #    forum.mark_read(request.user)
```
Comment (Lanny) [inline ISS/views/forum.py]: Why was this commented out?

File: ISS/views/forum.py [hunk 2]
```diff
@@ -90,6 +91,9 @@ def thread(request, thread_id):
     reply_form = _get_new_post_form(request)(author=request.user,
                                              initial={ 'thread': thread })
 
+    if thread.forum.member_view_only and not request.user.is_authenticated:
```
Comment (Lanny) [inline ISS/views/forum.py]: This check should happen before we bother the fetch posts a few lines above.

File: ISS/views/forum.py [hunk 3]
```diff
@@ -90,6 +91,9 @@ def thread(request, thread_id):
...
+        return render(request, 'members_only.html')
```
Comment (Lanny) [inline ISS/views/forum.py]: I like the specific page telling them what happened as opposed to the generic 403 page, but we should still answer with a 403 status code

**Comment by Lanny** [comment]
This didn't need to be closed, the code that's there is good, there just needs to be some additional filtering in other views.

**Comment by sometimesok** [comment]
yup, working on it.

**Pull request closed without merging**